app/controllers/schools/standings_controller.rb

```diff
@@ -0,0 +1,65 @@
+module Schools
+  class StandingsController < SchoolsController
+    layout "school"
+
+    # GET /school/standings/new
+    def new
+      @standing = Standing.new
+      authorize(@standing, policy_class: Schools::StandingPolicy)
+    end
+
+    # GET /school/standings/:id/edit
+    def edit
+      @standing = current_school.standings.find(params[:id])
+      authorize(@standing, policy_class: Schools::StandingPolicy)
+    end
+
+    # POST /school/standings
+    def create
+      @standing = Standing.new
+      authorize(@standing, policy_class: Schools::StandingPolicy)
+
+      standing_params =
+        params.require(:standing).permit(:name, :color, :description)
+
+      begin
+        @standing = current_school.standings.create!(standing_params)
+        flash[:success] = I18n.t("schools.standings.create.success")
+        redirect_to standing_school_path
+      rescue ActiveRecord::RecordInvalid => e
+        flash.now[:error] = e.message
+        render :new
+      end
+    end
+
+    # PATCH /school/standings/:id
+    def update
+      @standing = current_school.standings.find(params[:id])
+      authorize(@standing, policy_class: Schools::StandingPolicy)
+
+      standing_params =
+        params.require(:standing).permit(:name, :color, :description)
+      @standing.update!(standing_params)
+
+      flash[:success] = I18n.t("schools.standings.update.success")
+
+      redirect_to standing_school_path
+    end
+
+    # DELETE /school/standings/:id
+    def destroy
+      @standing = current_school.standings.find(params[:id])
+      authorize(@standing, policy_class: Schools::StandingPolicy)
+
+      if @standing.user_standings.count > 0
+        @standing.update!(archived_at: Time.zone.now)
+      else
+        @standing.destroy!
+      end
+
+      flash[:success] = I18n.t("schools.standings.delete.success")
```

Comment:
> We could have shown the warning clearly to the user and removed it!)

Currently, when deleting a standing with archive and delete, we show a count of the logs associated with a standing on confirm window

> Why did we pick archive over delete?

We will show the archived standings of the school on the other page in V2(We might need the data of the archived standings). A user standing is always associated with a standing, If we use choose to delete the standing when logs are associated then we need to duplicate the standing data (color, name) and store in the user standing table.



 > Having 2 behaviours would create confusion over time;
 
 The delete is done only when no user standing logs are associated with a standing. Just to avoid orphan standing records.
 
 > If we are preserving the behavior it would be good to show different error messages based on the action it performs.
 
 Ok, will think about it.
 
 Currently, on the coach notes page, if show a delete notification, on backend what we do is archive.

Reply:
In coach note we are always archiving, in this case we are archiving and deleting so i think we should either unify the behaviour or make the notifications clear.

Reply:
Fixed https://github.com/pupilfirst/pupilfirst/pull/1468/commits/43326341b933cc8d461434782edb61cfec7ab189

Showing a clear notification.